packages/action-button/src/spectrum-action-button.css

```diff
@@ -76,7 +76,7 @@ governing permissions and limitations under the License.
             var(--spectrum-actionbutton-focus-ring-gap)
     );
 }
-.spectrum-ActionButton--sizeXS {
+:host([size='xs']) {
```

Comment:
This is the main guts of making this possible.

The above in `action-button.css` is based on _our_ choice to manage child Icon sizes from the Action Button rather than require users to add a `size` attribute to both.

Reply:
Gotcha, makes sense. So in Spectrum CSS, action button sizes and child icon sizes are unrelated / you'd have to specify `--sizeXS` on both in order to get an XS icon within an XS button.